### ISSUE TITLE
Don't create multiple records for the same content.

### ISSUE THREAD

**Issue by nejdetckenobi** [oae/gnome-shell-pano#324]
Most people hit copy command several times.
That creates multiple records in Pano.

If new content is the same as one of the existing records, don't create another record. Move the existing record to the top of the list.

SHA256 might help while comparing two records.

You may want to make this feature optional.

**Comment by NinZeige** [comment]
It sounds like your requirements are already met in the current version. You can check if the logic of the `findOrCreateDbItem` function fits your needs [here](https://github.com/oae/gnome-shell-pano/blob/d0d71f8015ada661f35a1390502c8e09fa2a10b6/src/utils/panoItemFactory.ts#L123)

I haven’t encountered any issues with duplicate data. Copying the same content would just bring the record to the front. If you did face this issue, maybe I could take a look at the version of your extension?

**Comment by Totto16** [comment]
> It sounds like your requirements are already met in the current version. You can check if the logic of the `findOrCreateDbItem` function fits your needs [here](https://github.com/oae/gnome-shell-pano/blob/d0d71f8015ada661f35a1390502c8e09fa2a10b6/src/utils/panoItemFactory.ts#L123)
> 
> I haven’t encountered any issues with duplicate data. Copying the same content would just bring the record to the front. If you did face this issue, maybe I could take a look at the version of your extension?

I completely agree with you, it already does that, if it doesn't do that it is a bug, but I never encountered that with recent versions. Maybe if you worked with latex or `\\` heavy files, there was an error in escaping, which made the comparison fail sometimes, but it was fixed with https://github.com/oae/gnome-shell-pano/pull/266

**Comment by nejdetckenobi** [comment]
It was an escaping error. Not latex but some files with bunch of `\` characters. I updated the extension. Looks good now. Since there is no way to reproduce it anymore, with your permission, I'll close the issue.
